src/lib/nft/Base64Storage.js

```diff
@@ -0,0 +1,59 @@
+import { memoize } from 'lodash'
+
+import Config from '../../config/config'
+import { Blob, NFTStorage } from './client'
+
+class Base64Storage {
+  constructor(apiKey) {
+    this.client = new NFTStorage({ token: apiKey })
+  }
+
+  // eslint-disable-next-line require-await
+  async store(base64) {
+    const { client } = this
+    const blob = new Blob([base64])
+
+    return client.storeBlob(blob)
+  }
+
+  async load(cid, skipCache = false) {
+    const clear = () => this._clearCache(cid)
+
+    if (true === skipCache) {
+      clear()
+    }
+
+    try {
+      return await this._load(cid)
+    } catch (exception) {
+      clear()
+      throw exception
+    }
+  }
+
+  // eslint-disable-next-line require-await
+  async delete(cid) {
+    const { client } = this
+
+    return client.delete(cid)
+  }
+
+  _load = memoize(async cid => {
+    const url = `https://${cid}.ipfs.dweb.link`
```

Comment:
actually yes, there could be some CORS issues. it depends from the browser you're using and the ipfs to web gateway you chosen. i'm thinking also to add support for a few gateways (e.g. failed by some link -> fall back to the next one)

Reply:
I think it is a must since these are public gateways which we have no control of and don't know their uptime

> i'm thinking also to add support for a few gateways (e.g. failed by some link -> fall back to the next one)